### PR TITLE
Fix "not initialized" warnings

### PR DESCRIPTION
**Pull request by algodave**
I'm proposing this PR in order to avoid `instance variable not initialized` warnings

### CODE REVIEW

**Comment by ronwsmith** [comment]
Thanks for the contrib!  When do you see these errors occurring?

File: lib/billy/proxy.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def initialize
     def start(threaded = true)
       if threaded
         Thread.new { main_loop }
-        sleep(0.01) while @signature.nil?
+        sleep(0.01) while (not defined?(@signature)) or @signature.nil?
```
Comment (ronwsmith) [inline lib/billy/proxy.rb]: Let's do `||` here.

File: lib/billy/proxy_connection.rb [hunk 2]
```diff
@@ -35,7 +35,7 @@ def on_message_complete
       if @parser.http_method == 'CONNECT'
         restart_with_ssl(@parser.request_url)
       else
-        if @ssl
+        if defined?(@ssl) and @ssl
```
Comment (ronwsmith) [inline lib/billy/proxy_connection.rb]: Let's do `&&` here.

**Comment by algodave** [comment]
@ronwsmith said 

> Thanks for the contrib! When do you see these errors occurring?

I'm using `puffing-billy` configured this way:

```ruby

# in spec/support/billy.rb
Billy.configure do |c|
  c.cache = true
  c.persist_cache = true
end
Billy.config.logger.level = Logger::ERROR
Capybara.default_driver = :poltergeist_billy

# in my rspec contexts
      before do
        Billy.configure do |c|
          c.cache_path = 'spec/billy/my_context_path'
          c.merge_cached_responses_whitelist = [
            /useless-domain-one\.com/,
            /useless-domain-two\.com/
          ]
        end
      end

```

**Pull request merged**